factory/runType/runTypeComponentsHandler.go

```diff
@@ -236,6 +236,9 @@ func (mrc *managedRunTypeComponents) CheckSubcomponents() error {
 	if check.IfNil(mrc.shardMessengerFactory) {
 		return errors.ErrNilBroadCastShardMessengerFactoryHandler
 	}
+	if check.IfNil(mrc.shardMessengerFactory) {
```

Comment:
check for exportHandlerFactoryCreator

is the CheckSubComponentsMethod not used? Nothing failed on this check.

Reply:
Right, good catch;
It is not tested,  there is just a TODO for it